### ISSUE TITLE
[Config Support]: Camera works when manually recorded via FFMPEG, but in Frigate gives "Invalid data found when processing input"

### ISSUE THREAD

**Issue by DylanDonkersgoed** [blakeblackshear/frigate#9249]
### Describe the problem you are having

The FFMPEG process spawned by Frigate apparently crashes immediately after Frigate starts with the error:

>rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing 

From what I can tell from other GitHub issues this is a generic error message with many possible causes. If I record the same stream from within the docker container using:

```bash
ffmpeg -i 'rtsp://admin:OMITTED@cam1.lan/cam/realmonitor?channel=1&subtype=1' -c copy /media/frigate/tmp.mp4
```

it produces a viewable video with no errors.

My config used to work but broke after an update some time ago. It was many months ago (perhaps more than a year?) and I've been coming back to it on and off since trying to fix it, so I don't remember exactly which update. Sorry.

I noticed something else very strange looking at the web interface today. Frigate actually did register several events and record videos December 28th 2023 between 7:00PM and 11:45PM EST. These are the only events currently in it's history. I have no idea why it worked only for that short time, I don't think I made any changes around then.

### Version

0.12.1-367d724

### Frigate config file

```yaml
mqtt:
  enabled: false

cameras:
  amcam1:
    ffmpeg:
      inputs:
        - path: 'rtsp://admin:OMITTED@cam1.lan:554/cam/realmonitor?channel=1&subtype=1'
          roles:
            - detect
    detect:
      enabled: False
      width: 640
      height: 480
```

### Relevant log output

```shell
2024-01-07 17:33:56.142808364  [INFO] Starting Frigate...
2024-01-07 17:33:59.257680240  [2024-01-07 17:33:59] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2024-01-07 17:33:59.269627923  [2024-01-07 17:33:59] peewee_migrate                 INFO    : Starting migrations
2024-01-07 17:33:59.695840237  [2024-01-07 17:33:59] peewee_migrate                 INFO    : There is nothing to migrate
2024-01-07 17:33:59.809364417  [2024-01-07 17:33:59] frigate.app                    INFO    : Output process started: 288
2024-01-07 17:33:59.809453528  [2024-01-07 17:33:59] detector.cpu                   INFO    : Starting detection process: 287
2024-01-07 17:33:59.812582803  [2024-01-07 17:33:59] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2024-01-07 17:33:59.812644603  [2024-01-07 17:33:59] frigate.app                    INFO    : Camera processor started for amcam1: 291
2024-01-07 17:33:59.816353343  [2024-01-07 17:33:59] frigate.app                    INFO    : Capture process started for amcam1: 292
2024-01-07 17:34:04.955947521  [2024-01-07 17:34:04] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:34:04.956027650  [2024-01-07 17:34:04] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:34:19.842821061  [2024-01-07 17:34:19] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:34:19.843062097  [2024-01-07 17:34:19] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:34:19.843295256  [2024-01-07 17:34:19] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:34:24.977405815  [2024-01-07 17:34:24] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:34:24.977529365  [2024-01-07 17:34:24] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:34:29.856152421  [2024-01-07 17:34:29] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:34:29.856647306  [2024-01-07 17:34:29] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:34:29.856658435  [2024-01-07 17:34:29] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:34:34.991676544  [2024-01-07 17:34:34] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:34:34.991858249  [2024-01-07 17:34:34] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:34:39.866273437  [2024-01-07 17:34:39] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:34:39.866485419  [2024-01-07 17:34:39] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:34:39.866776445  [2024-01-07 17:34:39] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:34:45.006366074  [2024-01-07 17:34:45] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:34:45.006440641  [2024-01-07 17:34:45] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:34:49.877554464  [2024-01-07 17:34:49] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:34:49.877686686  [2024-01-07 17:34:49] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:34:49.877841928  [2024-01-07 17:34:49] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:34:55.032058848  [2024-01-07 17:34:55] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:34:55.032292940  [2024-01-07 17:34:55] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:34:59.886135807  [2024-01-07 17:34:59] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:34:59.886658711  [2024-01-07 17:34:59] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:34:59.886671220  [2024-01-07 17:34:59] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:05.018828625  [2024-01-07 17:35:05] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:05.018950038  [2024-01-07 17:35:05] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:09.897747620  [2024-01-07 17:35:09] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:09.898187516  [2024-01-07 17:35:09] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:09.898200085  [2024-01-07 17:35:09] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:15.038514024  [2024-01-07 17:35:15] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:15.038520660  [2024-01-07 17:35:15] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:19.909313862  [2024-01-07 17:35:19] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:19.909864856  [2024-01-07 17:35:19] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:19.909877490  [2024-01-07 17:35:19] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:25.048066847  [2024-01-07 17:35:25] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:25.048261207  [2024-01-07 17:35:25] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:29.919230365  [2024-01-07 17:35:29] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:29.919440849  [2024-01-07 17:35:29] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:29.919703757  [2024-01-07 17:35:29] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:35.056905199  [2024-01-07 17:35:35] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:35.057090212  [2024-01-07 17:35:35] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:39.931984559  [2024-01-07 17:35:39] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:39.932508270  [2024-01-07 17:35:39] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:39.932520528  [2024-01-07 17:35:39] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:45.074437364  [2024-01-07 17:35:45] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:45.074752704  [2024-01-07 17:35:45] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:49.942637641  [2024-01-07 17:35:49] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:49.943176522  [2024-01-07 17:35:49] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:49.943188961  [2024-01-07 17:35:49] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:35:55.120734580  [2024-01-07 17:35:55] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:35:55.121197959  [2024-01-07 17:35:55] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:35:59.955546147  [2024-01-07 17:35:59] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:35:59.955807969  [2024-01-07 17:35:59] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:35:59.956042228  [2024-01-07 17:35:59] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:36:05.112216449  [2024-01-07 17:36:05] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:36:05.112488009  [2024-01-07 17:36:05] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:36:09.966016414  [2024-01-07 17:36:09] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:36:09.966304212  [2024-01-07 17:36:09] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:36:09.971048110  [2024-01-07 17:36:09] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:36:15.107577439  [2024-01-07 17:36:15] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:36:15.107760294  [2024-01-07 17:36:15] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
2024-01-07 17:36:19.975236114  [2024-01-07 17:36:19] watchdog.amcam1                ERROR   : Ffmpeg process crashed unexpectedly for amcam1.
2024-01-07 17:36:19.975407884  [2024-01-07 17:36:19] watchdog.amcam1                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2024-01-07 17:36:19.975522377  [2024-01-07 17:36:19] ffmpeg.amcam1.detect           ERROR   : rtsp://*:*@cam1.lan:554/cam/realmonitor?channel=1&subtype=1: Invalid data found when processing input
2024-01-07 17:36:25.115017005  [2024-01-07 17:36:25] frigate.video                  ERROR   : amcam1: Unable to read frames from ffmpeg process.
2024-01-07 17:36:25.115221059  [2024-01-07 17:36:25] frigate.video                  ERROR   : amcam1: ffmpeg process is not running. exiting capture thread...
```

### Frigate stats

```json
{"amcam1":{"camera_fps":0.0,"capture_pid":292,"detection_enabled":0,"detection_fps":0.0,"ffmpeg_pid":396,"pid":291,"process_fps":0.0,"skipped_fps":0.0},"cpu_usages":{"%Cpu(s):":{"cpu":"id,","mem":"4.4"},"1":{"cpu":"0.0","mem":"0.0"},"101":{"cpu":"0.0","mem":"0.2"},"108":{"cpu":"0.0","mem":"0.0"},"112":{"cpu":"1.0","mem":"1.9"},"119":{"cpu":"0.0","mem":"0.2"},"134":{"cpu":"0.0","mem":"0.1"},"135":{"cpu":"0.0","mem":"0.1"},"136":{"cpu":"0.0","mem":"0.1"},"137":{"cpu":"0.0","mem":"0.1"},"15":{"cpu":"0.0","mem":"0.0"},"16":{"cpu":"0.0","mem":"0.0"},"24":{"cpu":"0.0","mem":"0.0"},"25":{"cpu":"0.0","mem":"0.0"},"26":{"cpu":"0.0","mem":"0.0"},"27":{"cpu":"0.0","mem":"0.0"},"28":{"cpu":"0.0","mem":"0.0"},"281":{"cpu":"0.0","mem":"1.0"},"286":{"cpu":"0.0","mem":"0.1"},"287":{"cpu":"0.0","mem":"1.1"},"288":{"cpu":"0.0","mem":"1.1"},"29":{"cpu":"0.0","mem":"0.0"},"291":{"cpu":"0.0","mem":"1.1"},"292":{"cpu":"0.3","mem":"1.0"},"297":{"cpu":"0.0","mem":"0.2"},"299":{"cpu":"0.0","mem":"0.2"},"30":{"cpu":"0.0","mem":"0.0"},"31":{"cpu":"0.0","mem":"0.0"},"32":{"cpu":"0.0","mem":"0.0"},"389":{"cpu":"0.0","mem":"0.0"},"400":{"cpu":"0.0","mem":"0.0"},"405":{"cpu":"0.0","mem":"0.2"},"41":{"cpu":"0.0","mem":"0.0"},"42":{"cpu":"0.0","mem":"0.0"},"80":{"cpu":"0.0","mem":"0.0"},"81":{"cpu":"0.0","mem":"0.0"},"82":{"cpu":"0.0","mem":"0.0"},"MiB":{"cpu":"3828.6","mem":"avail"},"PID":{"cpu":"%CPU","mem":"%MEM"},"Tasks:":{"cpu":"stopped,","mem":"0"},"top":{"cpu":"average:","mem":"0.38,"}},"detection_fps":0.0,"detectors":{"cpu":{"detection_start":0.0,"inference_speed":10.0,"pid":287}},"service":{"last_updated":1704667020,"latest_version":"0.12.1","storage":{"/dev/shm":{"free":66.8,"mount_type":"tmpfs","total":67.1,"used":0.3},"/media/frigate/clips":{"free":2502758.0,"mount_type":"nfs4","total":3936819.9,"used":1234006.2},"/media/frigate/recordings":{"free":2502758.0,"mount_type":"nfs4","total":3936819.9,"used":1234006.2},"/tmp/cache":{"free":1000.0,"mount_type":"tmpfs","total":1000.0,"used":0.0}},"temperatures":{},"uptime":181,"version":"0.12.1-367d724"}}
```

### Operating system

Other Linux

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Any other information that may be helpful

Frigate is running in Docker on Ubuntu 20.04 on an i5-4590 Optiplex desktop using Intel graphics.

I have reverted my Frigate config to the config recommended at https://docs.frigate.video/guides/getting_started/ for initial testing, but it is not my normal config.

My docker-compose.yml is below:

```yaml
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    shm_size: "64mb" # update for your cameras based on calculation above
    devices:
      # - /dev/bus/usb:/dev/bus/usb # passes the USB Coral, needs to be modified for other versions
      # - /dev/apex_0:/dev/apex_0 # passes a PCIe Coral, follow driver instructions here https://coral.ai/docs/m2/get-started/#2a-on-linux
      - /dev/dri/renderD128:/dev/dri/renderD128
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./frigate/config.yml:/config/config.yml:ro
      - /mnt/files/_Camera:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - 5000:5000
    networks:
      - internal
    environment:
      FRIGATE_RTSP_PASSWORD: "OMITTED"
```

Here is a screenshot of the settings from my camera, the one on the right is the stream currently configured but neither one works:
![amcrest-camera-settings](https://github.com/blakeblackshear/frigate/assets/7719733/49f92d3a-1f04-44d2-9b4e-5395ef5bdc11)

**Comment by NickM-27** [comment]
The error likely has nothing to do with recording, but with the decoding of the stream for motion and object detection. Is there a reason why you are trying to record and run detection on the low quality stream? Usually recordings are done on the high quality stream.

**Comment by NickM-27** [comment]
I would suggest keeping them separate and letting the logs run for longer then paste them

**Comment by DylanDonkersgoed** [comment]
I transferred the docker setup for this to a new host machine and now it's working.

I did switch it from connecting to the camera by hostname to connecting by IP. I don't _think_ that was the issue though - the errors in the logs this time around clearly indicated a connection issue. And I believe I tested connecting by IP before in any case. But that's the only thing I changed in the actual frigate/docker config.
